test/integ/pandas/test_arrow_pandas.py

```diff
@@ -1202,6 +1202,89 @@ def test_batch_to_pandas_arrow(conn_cnx, result_format):
+                pyarrow.bool_(),
+                pyarrow.string(),
+                pyarrow.string(),
+                pyarrow.list_(pyarrow.float64(), 5),
+            )
+
+            if enable_structured_types:
+                for param in {
+                    "ENABLE_STRUCTURED_TYPES_IN_CLIENT_RESPONSE",
+                    "IGNORE_CLIENT_VESRION_IN_STRUCTURED_TYPES_RESPONSE",
+                    "FORCE_ENABLE_STRUCTURED_TYPES_NATIVE_ARROW_FORMAT",
+                }:
+                    cur.execute(f"alter session set {param}=true")
+                expected_types += (
+                    pyarrow.map_(pyarrow.string(), pyarrow.int64()),
+                    pyarrow.struct(
+                        {"city": pyarrow.string(), "population": pyarrow.float64()}
+                    ),
+                    pyarrow.list_(pyarrow.float64()),
+                )
+            else:
+                expected_types += (
+                    pyarrow.string(),
+                    pyarrow.string(),
+                    pyarrow.string(),
+                )
+
+            # Ensure an empty batch to use default typing
+            # Otherwise arrow will resize types to save space
+            cur.execute(
+                """
+            select
+            1 :: INTEGER as FIXED_type,
+            2.0 :: FLOAT as REAL_type,
+            'test' :: TEXT as TEXT_type,
+            '2024-02-28' :: DATE as DATE_type,
+            '2020-03-12 01:02:03.123456789' :: TIMESTAMP as TIMESTAMP_type,
+            '{"foo": "bar"}' :: VARIANT as VARIANT_type,
+            '2020-03-12 01:02:03.123456789' :: TIMESTAMP_LTZ as TIMESTAMP_LTZ_type,
+            '2020-03-12 01:02:03.123456789' :: TIMESTAMP_TZ as TIMESTAMP_TZ_type,
+            '2020-03-12 01:02:03.123456789' :: TIMESTAMP_NTZ as TIMESTAMP_NTZ_type,
+            '0xAAAA' :: BINARY as BINARY_type,
+            '01:02:03.123456789' :: TIME as TIME_type,
+            true :: BOOLEAN as BOOLEAN_type,
+            TO_GEOGRAPHY('LINESTRING(13.4814 52.5015, -121.8212 36.8252)') as GEOGRAPHY_typejk,
+            TO_GEOMETRY('LINESTRING(13.4814 52.5015, -121.8212 36.8252)') as GEOMETRY_type,
+            [1,2,3,4,5] :: vector(float, 5) as VECTOR_type,
+            object_construct('k1', 1, 'k2', 2, 'k3', 3, 'k4', 4, 'k5', 5) :: map(varchar, int) as MAP_type,
+            object_construct('city', 'san jose', 'population', 0.05) :: object(city varchar, population float) as OBJECT_type,
+            [1.0, 3.1, 4.5] :: array(float) as ARRAY_type
+            WHERE 1=0
+            """
+            )
+
+            batches = cur.get_result_batches()
+            assert len(batches) == 1
+            batch = batches[0]
+            arrow_table = batch.to_arrow()
+            for actual, expected in zip(arrow_table.schema, expected_types):
+                assert (
+                    actual.type == expected
+                ), f"Expected {actual.name} :: {actual.type} column to be of type {expected}"
```

Comment:
would it be a good idea to also match data values along with types

Reply:
See my comment on Adam's comment.